util/common.go

```diff
@@ -37,23 +37,16 @@ func stringMapKeys(m map[string]*yang.Entry) []string {
 	return out
 }
 
-// stringMapSetToSlice converts a string set expressed as a map m, into a slice
-// of strings.
-func stringMapSetToSlice(m map[string]interface{}) []string {
-	var out []string
-	for k := range m {
-		out = append(out, k)
-	}
-	return out
-}
-
 // TODO(mostrowski): move below functions into path package.
 
 // pathMatchesPrefix reports whether prefix is a prefix of path.
 func pathMatchesPrefix(path *gpb.Path, prefix []string) bool {
 	if len(path.GetElem()) < len(prefix) {
 		return false
 	}
+	for len(prefix) != 0 && prefix[len(prefix)-1] == "" {
```

Comment:
Should we treat this case as an error -- i.e., I read this to be `prefix := []string{"one", "" , ""}` -- isn't the case that we have "" in the path something erroneous?

Reply:
I'm not sure. I haven't seen anything to suggest that // is illegal in schema paths so I've stuck with the common accepted path behavior. It sounds like you're interested in detecting errors with this - but perhaps those errors would be caught anyway at another layer without introducing this constraint here? Is there a specific scenario you can think of maybe?

Reply:
But isn't the only case that this check is really required based on the fact that you're using `strings.Split` on the input from tests? I'm just not sure that this mutation of the path is really the right thing to be doing regardless of whether this is an error or not.

Reply:
It's true that the reason trailing / appear is because they are coming from the tests, but that's exactly the point of the tests - I was to make sure that introducing a trailing / in the prefix is not going to lead to pathMatchesPrefix("a/b/c", "a/") failing. Although strictly speaking the trailing / should not be there I like having the extra robustness since trailing / often find their way into paths. 
I'm not sure if this is what you meant - if not let's chat live.

Reply:
No, I missed the point here -- thanks for clarifying for me.